frontend/packages/kubevirt-plugin/src/components/modals/cdrom-vm-modal/helpers.ts

```diff
@@ -8,3 +30,88 @@ export const getAvailableCDName = (cds: CD[]) => {
   }
   return `cd-drive-${index}`;
 };
+
+const getCDVMIDataVolumes = (
+  cds: CD[],
+  vm: VMILikeEntityKind,
+  dataVolumes?: K8sResourceKind[],
+): { [name: string]: K8sResourceKind } => {
+  if (dataVolumes && vm.kind !== VirtualMachineModel.kind) {
+    return Object.assign(
+      {},
+      ...cds.map(({ name }) => {
+        const dvName = getVMIDataVolumeNameByDisk(vm as VMIKind, name);
+        const dvObj = dataVolumes.find((dv) => dv.metadata.name === dvName);
+
+        return { [name]: dvObj };
+      }),
+    );
+  }
+
+  return Object.assign(
+    {},
+    ...cds.map(({ name }) => {
+      return { [name]: null };
+    }),
+  );
+};
+
+export const mapCDsToSource = (
```

Comment:
this should not be necessary and should be generic so it can be used by disks as well.

Please take a look at `useOwnedVolumeReferencedResources` and `getVolumeReferencedObject`.

Reply:
I don't understand. is it needed and should be generic or not needed at all ?

Reply:
not sure about cdroms as per discussions bellow. But for sure it will be needed for disks. 

Anyway it can be made generic and more correct with the approach mentioned above.

Reply:
sorry for the confusion, but we don't have old version of DVs or other resource which we could use. So we should drop diffing of other resources functionality.

Unless there is a strong use case for this, then we could implement our own history mechanism.